### DISCUSSION TITLE
Simulation Events

### DISCUSSION THREAD

**Discussion by weisselbiad** [manoelcampos/cloudsim-plus#387] [Q&A] [answered]
Hello,
I created my own Datacenter and Broker to provide GPU capabilities but i have a problem with managing SimEvents. 
After initializing the simulation there are some created events which take my new created Datacenter and broker as src and DatacenterSimple as a destination or target and vis-ver-ca which causes logical problem.
How can i avoid that and thanks )

**Comment by manoelcampos** [comment, marked as answer]
You cannot avoid Datacenter, Broker and other entities to receive messages.
If you mess with the messages already been sent, your simulation may not work or produce incorrect/invalid results.
If you want to extend such entities and need to pass additional messages, you have to create new message types in CloudSimTag class, then send and process such messages.

If you have downloaded the CloudSim Plus source code and is changing it without any intention to provide new features to the project, you probably are doing it the wrong way (as almost all projects using CloudSim). 
You have to create an empty maven project, include CloudSim Plus as a dependency, and then create the classes you want to extend, without touching CloudSim Plus code.

That is what a framework is used for: you don't mess with the framework code to create your own app.

**Comment by manoelcampos** [comment]
I've checked you create a standalone repo at https://github.com/weisselbiad/cloudsim-plus-modules. This is even more inappropriate. Your repo will become out-of-date when the framework is updated (which used to happen at a high frequency). Updating your copy to the new version may become a nightmare.

For instance, that version you copied is already completely out-of-date, has a totally different structure, and will be impractical to try to update to the new version. Modules were split into different repositories and a new version of the examples project was already released.
